vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -54,16 +67,31 @@ def parse_node(node)
 
     def parse_service(service)
       new_result = parse_base_item(service)
+      container_groups = []
+
+      endpoint_container_groups = @data_index.fetch_path(
+        :container_endpoints, :by_namespace_and_name, new_result[:namespace],
+        new_result[:name], :container_groups)
+
+      unless endpoint_container_groups.nil?
+        endpoint_container_groups.each do |group|
+          cg = @data_index.fetch_path(
+            :container_groups, :by_namespace_and_name, group[:namespace],
+            group[:name])
+          container_groups << cg unless cg.nil?
+        end
```

Comment:
And the nil can be avoided by defaulting endpoint_container_groups to an empty array:

```ruby
endpoint_container_groups = @data_index.fetch_path(
  :container_endpoints, :by_namespace_and_name, new_result[:namespace],
  new_result[:name], :container_groups)
endpoint_container_groups ||= []

container_groups = endpoint_container_groups.collect do |group|
  @data_index.fetch_path(
    :container_groups, :by_namespace_and_name, group[:namespace],
    group[:name])
end.compact
```